File: src/main/java/ua/softgroup/matrix/server/supervisor/producer/json/UserProjectTimeResponse.java
package ua.softgroup.matrix.server.supervisor.producer.json;

/**
 * @author Oleksandr Tyshkovets <sg.olexander@gmail.com>
 */
public class UserProjectTimeResponse extends TimeResponse {

    private long entityId;
    private String entityType = "project";

    public UserProjectTimeResponse() {
    }

    public UserProjectTimeResponse(long entityId, int workSeconds, int idleSeconds, double idlePercentage) {
        this.entityId = entityId;
        this.totalWorkTimeSeconds = workSeconds;
        this.totalIdleTimeSeconds = idleSeconds;
        this.totalIdlePercentage = idlePercentage;
    }

    public long getEntityId() {
        return entityId;
    }

    public void setEntityId(long entityId) {
        this.entityId = entityId;
    }

    public String getEntityType() {
        return entityType;
    }

    public void setEntityType(String entityType) {
        this.entityType = entityType;
    }
}

File: src/main/scala/ua/softgroup/matrix/server/supervisor/producer/json/UserProjectTimeResponse.scala
package ua.softgroup.matrix.server.supervisor.producer.json

import scala.beans.BeanProperty

/**
  * @author Oleksandr Tyshkovets <olexandr.tyshkovets@gmail.com>
  */
class UserProjectTimeResponse extends TimeResponse {

  @BeanProperty
  var entityId: Long = _

  @BeanProperty
  var entityType: String = "project"

  def this(entityId: Long, workSeconds: Int, idleSeconds: Int, idlePercentage: Double) {
    this()
    this.entityId = entityId
    this.totalWorkTimeSeconds = workSeconds
    this.totalIdleTimeSeconds = idleSeconds
    this.totalIdlePercentage = idlePercentage
  }

  override def toString: String = {
    "UserProjectTimeResponse(entityId=%d, totalWorkTimeSeconds=%d, totalIdleTimeSeconds=%d, totalIdlePercentage=%f)"
      .format(entityId, totalWorkTimeSeconds, totalIdleTimeSeconds, totalIdlePercentage)
  }
}
